src/utils/hooks/useConsoleNamespaceBookmarks/useConsoleNamespaceBookmarks.ts

```diff
@@ -0,0 +1,147 @@
+import { useEffect, useState } from 'react';
+
+import {
+  ConfigMapModel,
+  modelToGroupVersionKind,
+  UserModel,
+} from '@kubevirt-ui/kubevirt-api/console';
+import { IoK8sApiCoreV1ConfigMap } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import useK8sWatchData from '@multicluster/hooks/useK8sWatchData';
+import { kubevirtK8sCreate, kubevirtK8sPatch } from '@multicluster/k8sRequests';
+
+const CONSOLE_USER_SETTINGS_NAMESPACE = 'openshift-console-user-settings';
+const CONSOLE_USER_SETTINGS_CONFIG_MAP_PREFIX = 'user-settings-';
+const CONSOLE_NAMESPACE_BOOKMARKS_KEY = 'console.namespace.bookmarks';
+
+type ConsoleNamespaceBookmarks = Record<string, boolean>;
+
+type UseConsoleNamespaceBookmarks = (
+  cluster?: string,
+) => [
+  ConsoleNamespaceBookmarks,
+  (bookmarks: ConsoleNamespaceBookmarks) => Promise<ConsoleNamespaceBookmarks>,
+  boolean,
+  Error | undefined,
+];
+
+const useConsoleNamespaceBookmarks: UseConsoleNamespaceBookmarks = (cluster) => {
+  const [error, setError] = useState<Error>();
+  const [bookmarks, setBookmarks] = useState<ConsoleNamespaceBookmarks>({});
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const [user, loadedUser, errorUser] = useK8sWatchData<IoK8sApiCoreV1ConfigMap>({
+    cluster,
+    groupVersionKind: modelToGroupVersionKind(UserModel),
+    name: '~',
+  });
+
+  const userName = user?.metadata?.uid || user?.metadata?.name?.replace(/[^-._a-zA-Z0-9]+/g, '-');
+  const configMapName = userName ? `${CONSOLE_USER_SETTINGS_CONFIG_MAP_PREFIX}${userName}` : null;
+
+  const [userConfigMap, loadedConfigMap, configMapError] = useK8sWatchData<IoK8sApiCoreV1ConfigMap>(
+    configMapName
+      ? {
+          cluster,
+          groupVersionKind: modelToGroupVersionKind(ConfigMapModel),
+          name: configMapName,
+          namespace: CONSOLE_USER_SETTINGS_NAMESPACE,
+        }
+      : null,
+  );
```

Comment:
i would create a new hook to get the user configmap and use it here and in the `useConsoleClusterBookmarks` but we can do it in another pr